### PR TITLE
feat: Support a single program indicators as filter [DHIS2-10908]

### PR DESCRIPTION
**Pull request by d-bernat**

### CODE REVIEW

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java [hunk 1]
```diff
@@ -147,7 +147,8 @@ else if ( params.hasStartEndDate() && !params.getReportingRates().isEmpty() )
         {
             error = new ErrorMessage( ErrorCode.E7107 );
         }
-        else if ( !params.getFilterIndicators().isEmpty() && params.getFilterOptions( DATA_X_DIM_ID ).size() > 1 )
+        else if ( (!params.getFilterIndicators().isEmpty() || !params.getFilterProgramIndicators().isEmpty())
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java:150]: Could we please add a unit test to verify the behavior in `DefaultQueryValidator`?

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java [hunk 2]
```diff
@@ -1983,25 +1992,26 @@ public boolean equals( Object object )
...
+            if ( !dimensions.equals( other.dimensions ) )
             {
                 return false;
             }
+            if ( filters == null )
+            {
+                return other.filters == null;
+            }
+            else
+                return filters.equals( other.filters );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:2013]: Add brackets { } even for one-liners per the code style.

Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:2016]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java:150]: Done

**Review by larshelge — APPROVED** [review]

**Pull request merged**